postgresql/resource_postgresql_event_trigger.go

```diff
@@ -0,0 +1,408 @@
+			},
+			eventTriggerFunctionSchemaAttr: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Schema where the function is located.",
+			},
+			eventTriggerStatusAttr: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "enable",
+				Description: "These configure the firing of event triggers. A disabled trigger is still known to the system, but is not executed when its triggering event occurs",
+				ValidateFunc: validation.StringInSlice([]string{
+					"disable",
+					"enable",
+					"enable_replica",
+					"enable_always",
+				}, false),
+			},
+			eventTriggerOwnerAttr: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The user name of the owner of the event trigger. You can't use 'current_role', 'current_user' or 'session_user' in order to avoid drifts",
+				ValidateFunc: validation.StringNotInSlice([]string{
+					"current_role",
+					"current_user",
+					"session_user",
+				}, true),
+			},
+		},
+	}
+}
+
+func resourcePostgreSQLEventTriggerCreate(db *DBConnection, d *schema.ResourceData) error {
+	eventTriggerName := d.Get(eventTriggerNameAttr).(string)
+	d.SetId(eventTriggerName)
+
+	b := bytes.NewBufferString("CREATE EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerOn := d.Get(eventTriggerOnAttr).(string)
+	fmt.Fprint(b, " ON ", eventTriggerOn)
+
+	if filters, ok := d.GetOk(eventTriggerFilterAttr); ok {
+		filters := filters.([]interface{})
+
+		for i, filter := range filters {
+			filter := filter.(map[string]interface{})
+
+			if variable, ok := filter[eventTriggerFilterVariableAttr]; ok {
+				if i == 0 {
+					fmt.Fprint(b, " WHEN ", variable)
+				} else {
+					fmt.Fprint(b, " AND ", variable)
+				}
+			}
+
+			if values, ok := filter[eventTriggerFilterValueAttr]; ok {
+				var new_values []string
+
+				for _, value := range values.([]interface{}) {
+					new_values = append(new_values, pq.QuoteLiteral(value.(string)))
+				}
+
+				fmt.Fprint(b, " IN (", strings.Join(new_values, ","), ")")
+			}
+		}
+	}
+
+	eventTriggerFunction := d.Get(eventTriggerFunctionAttr).(string)
+	eventTriggerSchema := d.Get(eventTriggerFunctionSchemaAttr).(string)
+	fmt.Fprint(b, " EXECUTE FUNCTION ", pq.QuoteIdentifier(eventTriggerSchema), ".", eventTriggerFunction, "()")
+
+	createSql := b.String()
+
+	// Enable or disable the event trigger
+	b = bytes.NewBufferString("ALTER EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerEnabled := d.Get(eventTriggerStatusAttr).(string)
+	fmt.Fprint(b, " ", eventTriggerEnabled)
+
+	statusSql := b.String()
+
+	// Table owner
+	b = bytes.NewBufferString("ALTER EVENT TRIGGER ")
+	eventTriggerOwner := d.Get(eventTriggerOwnerAttr).(string)
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName), " OWNER TO ", eventTriggerOwner)
```

Comment:
I guess we should quote the owner too?
```suggestion
	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName), " OWNER TO ", pq.QuoteIdentifier(eventTriggerOwner))
```